### PR TITLE
Add option for negative cumulative summation in LayeredBase. 

### LINKED ISSUE #19480: Allow reverse cumulative summation direction for LayeredBase
## Reason
The `LayeredBase` currently assumes that when `cumulative` is true, that you want to sum the layer values moving in the positive direction. For some Cardinal applications, we have flow moving in the negative-z direction, and we want to apply cumulative layer user objects in the same direction as the flow. However, the current assumption that you want to sum moving in the positive direction gives the wrong values.

## Design
Add a parameter like `positive_cumulative_direction` that indicates what direction (positive or negative along the `direction`) to assemble the cumulative values.

## Impact
New feature.

### PR DESCRIPTION
**Pull request by aprilnovak**
This adds the ability to sum the cumulative values in the "negative" direction in `LayeredBase`.

Closes #19480

### CODE REVIEW

File: framework/include/userobject/LayeredBase.h [hunk 1]
```diff
@@ -152,6 +152,9 @@ class LayeredBase : private Restartable
   /// Whether the values are cumulative over the layers
   bool _cumulative;
 
+  /// Whether the cumulative values should be summed in the positive or negative direction
+  bool _positive_cumulative_direction;
```
Comment (GiudGiud) [inline framework/include/userobject/LayeredBase.h:156]: const

**Review by GiudGiud — COMMENTED** [review]
Looks good.
Is there an interest for just making the axis be a general direction instead of X, Y, Z ?
I think -X, -Y, -Z would cover this

**Comment by aprilnovak** [comment]
> Is there an interest for just making the axis be a general direction instead of X, Y, Z ?

 I personally don't have a need for a general feature like that. Would you prefer I add `-x`, `-y`, and `-z` to the MooseEnum instead of the `positive_cumulative_direction` parameter? Can you have a `-` leading the name of a MooseEnum?

**Comment by GiudGiud** [comment]
Probably but I think I like what you did just as much.

File: framework/include/userobject/LayeredBase.h (hunk 1 above)
Comment (GiudGiud) [reply framework/include/userobject/LayeredBase.h:156]:
```suggestion
  const bool _positive_cumulative_direction;
```

File: framework/include/userobject/LayeredBase.h [hunk 2]
```diff
@@ -152,6 +152,9 @@ class LayeredBase : private Restartable
   /// Whether the values are cumulative over the layers
   bool _cumulative;
```
Comment (GiudGiud) [inline framework/include/userobject/LayeredBase.h:153]:
```suggestion
  const bool _cumulative;
```

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**